awspub/image_marketplace.py

```diff
@@ -43,11 +44,14 @@ def request_new_version(self, image_id: str) -> None:
 
         # version doesn't exist already - create a new one
         changeset = self._request_new_version_changeset(image_id)
+        # changeset names cannot include some characters valid for a version_title. Remove all invalid characters
+        changeset_name = re.sub(
```

Comment:
moving this to an extra method which can then be parametrized unit tested against multiple input strings would be nice.

Reply:
moved to staticmethod and parametrized tests